src/mongo/delegates/Views.class.php

```diff
@@ -19,7 +19,7 @@ class Views extends CompositeBase
      * Construct accepts actual objects rather than strings as this class is a delegate of
      * Tripod and should inherit connections set up there
      * @param string $storeName
-     * @param \MongoCollection $collection
+     * @param \MongoCollection $collection // TODO: should this really be a constructor arg?
```

Comment:
I can't fathom why this is a constructor argument.

`Views` is a class that is used to handle all views, and as such is not tied to a single collection.  Indeed, many of the operations operate directly on the `"views"` collection, rather than a CBD collection.  However, in order to construct a `Views` instance, you need to pass a CBD collection.  This seems to be overly coupled with the use of `Views` from a tripod instance, but it hinders other use cases.

The collection and associated podName are barely used in this class, and even then, only in `generateViews`, which is only invoked thus:

```php
public function update(\Tripod\Mongo\ImpactedSubject $subject)
    {
        $resource = $subject->getResourceId();
        $resourceUri    = $resource[_ID_RESOURCE];
        $context        = $resource[_ID_CONTEXT];

        $this->generateViews(array($resourceUri),$context);
    }
```

The `ImpactedSubject` has the podName, so it would be perfectly possible to fetch this collection and pass as an argument to `generateViews`.

How would y'all feel about me removing this constructor argument?